lib/core/services/api/api_service_impl.dart

```diff
@@ -76,6 +77,36 @@ class ApiServiceImpl extends ApiService {
     }
   }
 
+  @override
+  Future<void> facebookLogin() async {
+    try {
+      // Create an instance of FacebookLogin
+      final fb = FacebookLogin();
+
+      // Log in
+      final result = await fb.expressLogin();
+
+      if (result.status == FacebookLoginStatus.success) {
+        final FacebookAccessToken? accessToken = result.accessToken;
+        print(true);
+        print('Access token: ${accessToken?.token}');
+
+        final body = {
+          'provider': 'facebook',
+          'access_token': accessToken?.token,
+        };
+        final response = await dio.post('/social/login', data: body);
+
+        if (response.statusCode == 200 && response.data != null) {
+          print(true);
```

Comment:
Handle response user data from api and save to shared preferences to be used in profile page.